source/azure_iot_hub_client.c

```diff
@@ -0,0 +1,538 @@
+        az_span device_id_span = az_span_create( ( uint8_t * ) pucDeviceId, ( int32_t ) ulDeviceIdLength );
+
+        options = az_iot_hub_client_options_default();
+
+        if( pxHubClientOptions )
+        {
+            options.module_id = az_span_create( ( uint8_t * ) pxHubClientOptions->pucModuleID, ( int32_t ) pxHubClientOptions->ulModuleIDLength );
+            options.model_id = az_span_create( ( uint8_t * ) pxHubClientOptions->pucModelID, ( int32_t ) pxHubClientOptions->ulModelIDLength );
+            options.user_agent = az_span_create( ( uint8_t * ) pxHubClientOptions->pucUserAgent, ( int32_t ) pxHubClientOptions->ulUserAgentLength );
+        }
+        else
+        {
+            options.user_agent = az_span_create( ( uint8_t * ) azureiothubUSER_AGENT, sizeof( azureiothubUSER_AGENT ) - 1 );
+        }
+
+        if( az_result_failed( az_iot_hub_client_init( &xAzureIoTHubClientHandle->_internal.xAzureIoTHubClientCore,
+                                                      hostname_span, device_id_span, &options ) ) )
+        {
+            AZLogError( ( "Failed to initialize az_iot_hub_client_init \r\n" ) );
+            ret = eAzureIoTHubClientInitFailed;
+        }
+        /* Initialize AzureIoTMQTT library. */
+        else if( ( AzureIoTMQTT_Init( &( xAzureIoTHubClientHandle->_internal.xMQTTContext ), pxTransportInterface,
+                                                prvGetTimeMs, prvEventCallback,
+                                                pucBuffer, ulBufferLength ) ) != AzureIoTMQTTSuccess )
+        {
+            AZLogError( ( "Failed to initialize AzureIoTMQTT_Init \r\n" ) );
+            ret = eAzureIoTHubClientInitFailed;
+        }
+        else
+        {
+            xAzureIoTHubClientHandle->_internal.pucDeviceID = pucDeviceId;
+            xAzureIoTHubClientHandle->_internal.ulDeviceIDLength = ulDeviceIdLength;
+            xAzureIoTHubClientHandle->_internal.pucHostname = pucHostname;
+            xAzureIoTHubClientHandle->_internal.ulHostnameLength = ulHostnameLength;
+            xAzureIoTHubClientHandle->_internal.xAzureIoTHubClientTimeFunction = xGetTimeFunction;
+            ret = eAzureIoTHubClientSuccess;
+        }
+    }
+
+    return ret;
+}
+/*-----------------------------------------------------------*/
+
+void AzureIoTHubClient_Deinit( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle )
+{
+    ( void ) xAzureIoTHubClientHandle;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTHubClientResult_t AzureIoTHubClient_Connect( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                     bool cleanSession,
+                                                     uint32_t ulTimeoutMilliseconds )
+{
+    AzureIoTMQTTConnectInfo_t xConnectInfo = { 0 };
+    AzureIoTHubClientResult_t ret;
+    AzureIoTMQTTStatus_t xResult;
+    bool xSessionPresent;
```

Comment:
I think we need to plumb this up to the caller, don't we?  If the session isn't present they're going to need to behave differently (re-subscribe) than if session is there and they asked it to be.

Reply:
pardon the extra uncrustify stuff here but here's the change:
https://github.com/Azure/azure-iot-middleware-freertos/pull/60/commits/3da1d16544476821cb4ac269743012f53a89335b

@hihigupt @jspaith